docs/CHANGELOG.md

```diff
@@ -74,6 +74,10 @@ NOTE: AccessLists are part of the 0x2 transaction type spec and Chainlink also i
 
 Avalanche AP4 defaults have been added (you can remove manually set ENV vars controlling gas pricing).
 
+#### New env vars
+
+`CHAIN_TYPE` - Configure the type of chain (if not standard). `Arbitrum`, `ExChain`, `Optimism`, or `XDai`. Replaces `LAYER_2_TYPE`.
```

Comment:
Let's clarify here. Chain types enable chain-specific quirks, it is automatically set so nops almost never need to use this unless they want to specifically override for a particular chain ID.

This has made me think of another problem - if we want to force "optimism" mode on a chain with a new chain ID, none of the other default optimism config will apply either. How to handle that?

Reply:
Yeah without a mechanism to inherit the defaults it must be fully configured.

Reply:
It looks like I should update `(c *chainScopedConfig) validate()` with some `ChainType` specific logic. When the chain ID is known to us, I think we could probably just reject a 'wrong' `ChainType`.

Reply:
Validation and tests added.